static/app/components/actions/ignore.tsx

```diff
@@ -17,7 +17,7 @@ import {
   IgnoredStatusDetails,
   SelectValue,
 } from 'sentry/types';
-import {getDuration} from 'sentry/utils/formatters';
+import getDuration from 'sentry/utils/duration/getDuration';
```

Comment:
Any reason for changing these to default exports? I would rather keep named exports as they are future-proof to renaming.

Reply:
I hope we don't need to rename it, unless it gets a `deprecated` suffix, it's more likely to be deleted 🤞. Overall I'm suspicious of whether we endup in the future where a rename is the hardest part and there's only one way to make it easy. Another angle to look at is: are we making a choice to build things based on tooling capability? We do make those choices all that time (by choosing, eslint, prettier, typescript, and many more nuanced examples) and it's super beneficial! But those examples have other benefits and standardization of tools matters, in this case the tools for renaming don't need to be standardized on.

That's the anti side, but pro side for going in the other direction is to deter folks from adding another export to the file. Also, we've got files like `utils/formatters.tsx` and `utils/dates.tsx` which i find opaque, it's hard to know what's in there. So I wanted to make the functions more discoverable by giving them their own file, now the filetree can show me what's available to potentially reuse. In that world I like having the filename match the function name inside, and i hope we stay in the world where the file doesn't need to export a second function. Therefore, the deterrence.


not trying to sound ranty, really just a stream of thought here ^

Reply:
No worries, I understand (and agree) with the point of steering folks towards single exports, however I don't think default exports are the right way to achieve that and I'm afraid we'll inevitably end up with a mixed bag of exports default/named exports again.

>Overall I'm suspicious of whether we endup in the future where a rename is the hardest part and there's only one way to make it easy.

I dont think it will ever be, but changes are a natural part in the codebase lifecycle and if something is renamed or has it's signature changed, then it's usage should be updated. The risk of not updating the naming is that eventually, we end up in a world where we have different namings for the same thing across the codebase which impacts things like searching (you can't really reliably grep for exported name) or running code mods (same issue).

>Another angle to look at is: are we making a choice to build things based on tooling capability?

I dont think we are, named exports are (at least for me) are just about keeping the codebase tidy over time.

>That's the anti side, but pro side for going in the other direction is to deter folks from adding another export to the file.

Agree with the goal, but we can achieve this with named exports as well.

I think this is a good debate to have and something we can bring up at the TSC (least we will outline the pros/cons for folks) and eod, I think we are aligned on the end goal of making the codebase more discoverable which I very much welcome 🙏🏼

Reply:
FWIW, I'm very much #teamnamedexports. I'm all for splitting stuff into multiple files when appropriate (and holy cow could we stand to do more of that on the Python side), but there's also (no pun intended) utility in being able to see related functions in a single spot, rather than mixed in with other functions. (Yes, we could solve this with deeper folder structure, but that has its own drawbacks, including having to update all references to any files which end up in a new folder once there's enough related stuff to justify corralling it into its own folder). 

> The risk of not updating the naming is that eventually, we end up in a world where we have different namings for the same thing across the codebase which impacts things like searching (you can't really reliably grep for exported name) or running code mods (same issue).

This (the same function going by different names) is an issue for sure, but the opposite (different functions going by the same name) is also a problem. Having named exports prevents the situation where files `A` and `B` both have default exports, and `A` gets default-imported into one file as `doStuff` and `B` gets default-imported into another file as `doStuff`, and it takes scrolling to the top of the two respective files to discover it's not the same `doStuff` being called in each file. (Granted, both functions could have the same name as named exports, but at least then then can be namespaced as `module.functionName` to keep the two straight.)

Reply:
Yup, I totally get it. I suppose I was more or less just chiming in to offer my $.02, so that two of the people who will definitely be at the TSC (especially the one likely to be arguing the pro-named-exports side) have at least heard a point I thought was important but being missed, and it's been written down in at least one spot where this question is being considered - not so much saying "you should change this PR." Perhaps I could have been clearer about that, though, sorry.

Reply:
Let's capture this discussion over in https://github.com/getsentry/frontend-tsc/issues/33

Reply:
Good idea, @evanpurkhiser. Added my points above there.